src/shared_utilities/models.jl

```diff
@@ -266,26 +270,28 @@ Adjustments to this - for example because different auxiliary variables
 have different dimensions - require defining a new method.
 """
 function initialize_auxiliary(model::AbstractModel{FT}, state) where {FT}
-    initialize_vars(
+    p = initialize_vars(
         auxiliary_vars(model),
         auxiliary_types(model),
         state,
         name(model),
     )
+    # assumes model.domain exists. this breaks a tutorial. we should
+    # change that tutorial? 
+    p = add_dss_buffer_to_aux(p, model.domain)
+    return p
 end
 
 function initialize_vars(keys, types, state, model_name)
     FT = eltype(state)
     if length(keys) == 0
-        return ClimaCore.Fields.FieldVector(; model_name => FT[])
+        return (; model_name => FT[])
```

Comment:
we should check the requirements of a NamedTuple that gets converted to a Field Vector. To me it would make more sense to have `(; model_name => (;))` for type consistency with the `else` case, but I'm not sure if we're able to do that.

Reply:
That makes sense. I think we are not able though:
```
julia> a = (; :foo => Float32[])
(foo = Float32[],)

julia> b = (;)
NamedTuple()

julia> ClimaCore.Fields.FieldVector(;a)
1-blocked 0-element ClimaCore.Fields.FieldVector{Float32, NamedTuple{(:a,), Tuple{ClimaCore.Fields.FieldVector{Float32, NamedTuple{(:foo,), Tuple{Vector{Float32}}}}}}}

julia> ClimaCore.Fields.FieldVector(;b)
ERROR: ArgumentError: Union{} does not have elements
```